### PR TITLE
PVP-219

### PR DESCRIPTION
**Pull request by KhajiitLikeToSneak**
- Can sort friends by experience, points.
- Added icons to top 3 friends.
- Changed AlertDialog to ButtonWithDialog.

_Note: removed isRequestSent, because I would probably need to mess with Dialog component in Buttons.kt and I don't know if I should. So 'Add friend' dialog now closes regardless of the returned toast result. 
Previously dialog was closing only if toast was "Friend request sent!" or "Both of you want to be friends! Request accepted!"._

### CODE REVIEW

File: app/src/main/java/com/pvp/app/ui/screen/friends/FriendsScreen.kt [hunk 1]
```diff
@@ -104,199 +109,196 @@ fun FriendsScreen(
...
+                    )
+                },
+                contentAlignment = Alignment.Center,
+                contentPadding = PaddingValues(
+                    horizontal = 45.dp,
+                    vertical = 2.dp
+                ),
+                dialogTitle = { Text("Friend Requests") },
+                dialogContent = {
+                    val selectedTab = remember { mutableIntStateOf(0) }
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/friends/FriendsScreen.kt:171]: Separate

File: app/src/main/java/com/pvp/app/ui/screen/friends/FriendsScreen.kt [hunk 2]
```diff
@@ -104,199 +109,196 @@ fun FriendsScreen(
...
+                                selected = selectedTab.intValue == 0,
+                                onClick = { selectedTab.intValue = 0 }
+                            ) {
+                                Text(
+                                    "Received",
+                                    style = MaterialTheme.typography.titleMedium,
+                                    fontWeight = if (selectedTab.intValue == 0) FontWeight.Bold else FontWeight.Normal
+                                )
+                            }
+                            Tab(
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/friends/FriendsScreen.kt:184]: Separate

File: app/src/main/java/com/pvp/app/ui/screen/friends/FriendsScreen.kt [hunk 3]
```diff
@@ -104,199 +109,196 @@ fun FriendsScreen(
...
+                                    "Sent",
+                                    style = MaterialTheme.typography.titleMedium,
+                                    fontWeight = if (selectedTab.intValue == 1) FontWeight.Bold else FontWeight.Normal
+                                )
+                            }
                         }
-                    }
 
-                    Spacer(modifier = Modifier.height(6.dp))
+                        Spacer(modifier = Modifier.height(6.dp))
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/friends/FriendsScreen.kt:196]: Is this a tab symbol there? Use spaces. `Ctrl+Alt+L`

File: app/src/main/java/com/pvp/app/ui/screen/friends/FriendsScreen.kt [hunk 4]
```diff
@@ -104,199 +109,196 @@ fun FriendsScreen(
...
-                    when (selectedTab.intValue) {
-                        0 -> {
-                            RequestList(
-                                requests = receivedRequests,
-                                requestTitle = "received",
-                                acceptAction = { request -> model.acceptFriendRequest(request) },
-                                denyAction = { request -> model.denyFriendRequest(request) }
-                            )
-                        }
+                        when (selectedTab.intValue) {
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/friends/FriendsScreen.kt:198]: Same as above

File: app/src/main/java/com/pvp/app/ui/screen/friends/FriendsScreen.kt [hunk 5]
```diff
@@ -328,11 +330,41 @@ private fun FriendList(
...
+                        Icon(
+                            imageVector = Icons.Outlined.EmojiEvents,
+                            contentDescription = "Top 1",
+                            modifier = Modifier
+                                .padding(start = 4.dp)
+                                .size(24.dp)
+                        )
+                    }
+
+                    index == 1 || index == 2 -> {
```
Comment (klaidoshka) [inline app/src/main/java/com/pvp/app/ui/screen/friends/FriendsScreen.kt:344]: `<= 2` would give the same result, I believe.

**Review by klaidoshka — APPROVED** [review]

**Review by vytjar — APPROVED** [review]

**Pull request merged**